File: client/src/app/_services/account.service.ts
import { Observable } from 'rxjs';
import { Customer } from './../classes/customer';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Account } from '../classes/account';
import { environment } from 'src/environments/environment';


@Injectable({
  providedIn: 'root'
})
export class AccountService {

  private API_URL= environment.apiUrl;


  constructor(private http: HttpClient) { }


  getAccountsForUser(userId: string): Observable<any> {
    return this.http.get<Account[]>(this.API_URL + `/accounts/${userId}`, { observe: "response" });
  }


  addAccount(account: Account, userId: string): Observable<any> {
    return this.http.post<any>(this.API_URL + `/accounts/${userId}`, account);
  }

  getAccountById(accountId: string): Observable<any> {
    return this.http.get<Account>(this.API_URL + `/accounts/${accountId}`, { observe: "response" });
  }

  updateAccount(accountId: string, account: Account): Observable<any> {
    return this.http.put<Account>(this.API_URL + `/accounts/${accountId}`, account);
  }

  deleteAccountById(userId: string, accountId: string): Observable<any> {
    return this.http.delete<Account>(this.API_URL + `/accounts/${userId}/${accountId}`);
  }
}

File: server/src/main/java/com/djordje/service/AccountService.java
package com.djordje.service;

import com.djordje.shared.dto.AccountDTO;
import com.djordje.ui.model.request.AccountDetailsRequestModel;
import com.djordje.ui.model.response.AccountRest;

import java.util.List;

public interface AccountService {

    AccountDTO addAccount(AccountDTO accountDTO, String userId);

    List<AccountDTO> getUserAccounts(String id);

    AccountDTO getAccountById(String accountId);

    AccountDTO updateAccount(AccountDTO accountDTO, String accountId);

    List<AccountDTO> deleteAccountById(String userId, String accountId);
}
